### PR TITLE
remap_restarts.py: add command line option; move temporary directory to out_dir for merra2; add more test cases

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
This PR further enhances the new "remap_restarts.py" utility:
- Adds command line option for exercising "remap_restarts.py" in GEOSldas and GCM setup.
- Minor cleanup:  If needed, copy M-2 files from tape archive into out_dir, thereby avoiding use of separate temp dir.
- Adds more test cases (see also #265 and #267).  

Still missing: EASE-based test case, see https://github.com/GEOS-ESM/GMAO_Shared/issues/265#issuecomment-1192870149

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
@sdrabenh , I have tested this PR and it is working.

File: GEOS_Util/post/remap_restart/remap_questions.py [hunk 1]
```diff
@@ -101,18 +95,19 @@ def ask_questions():
...
+            "name": "output:shared:out_dir",
+            "message": "Enter the directory for new restarts:\n"
         },
 
         {
             "type": "path",
-            "name": "output:shared:out_dir",
-            "message": "Enter the directory for new restarts:\n"
+            "name": "input:shared:rst_dir",
+            "message": "Enter a temporary directory for archived MERRA-2 files: ",
```
Comment (gmao-rreichle) [inline GEOS_Util/post/remap_restart/remap_questions.py:108]: @weiyuan-jiang:  I think the "message" here is confusing.  In an earlier commit, it was changed to the following:
`"message": "Enter a directory to which the archived MERRA-2 archive files can be copied: ",`
Is there a reason for changing the message again?

Comment (weiyuan-jiang) [reply GEOS_Util/post/remap_restart/remap_questions.py:108]: Not really. I just want to make the message short.

Comment (weiyuan-jiang) [reply GEOS_Util/post/remap_restart/remap_questions.py]: I remove this question and hide this temporary folder from users

**Comment by weiyuan-jiang** [comment]
@sdrabenh @mathomp4 This PR is ready. Would you please review it ?

**Review by gmao-rreichle — APPROVED** [review]
Looks ok to me.

**Review by sdrabenh — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**